tests/test_quickbooks_interrupted_sync.py

```diff
@@ -0,0 +1,216 @@
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+        final_state = menagerie.get_state(conn_id)
+        currently_syncing = final_state.get('currently_syncing')
+
+        # Checking resuming the sync resulted in a successfully saved state
+        with self.subTest():
+
+            # Verify sync is not interrupted by checking currently_syncing in the state for sync
+            self.assertIsNone(currently_syncing)
+
+            # Verify bookmarks are saved
+            self.assertIsNotNone(final_state.get('bookmarks'))
+
+            # Verify final_state is equal to uninterrupted sync's state
+            # (This is what the value would have been without an interruption and proves resuming succeeds)
+            self.assertDictEqual(final_state, full_sync_state)
+
+        # Stream level assertions
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # Gather expectations
+                expected_replication_method = self.expected_replication_method()[stream]
+                expected_replication_key = next(iter(self.expected_replication_keys()[stream]))
+                
+                # Gather actual results
+                full_records = [message['data'] for message in synced_records_full_sync.get(stream, {}).get('messages', [])]
+                full_record_count = record_count_by_stream_full_sync.get(stream, 0)
+                interrupted_records = [message['data'] for message in synced_records_interrupted_sync.get(stream, {}).get('messages', [])]
+                interrupted_record_count = record_count_by_stream_interrupted_sync.get(stream, 0)
+
+                # Final bookmark after interrupted sync
+                final_stream_bookmark = final_state['bookmarks'][stream]['LastUpdatedTime']
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # Verify final bookmark saved the match formatting standards for resuming sync
+                    self.assertIsNotNone(final_stream_bookmark)
+                    self.assertIsInstance(final_stream_bookmark, str)
+                    self.assertIsDateFormat(final_stream_bookmark, "%Y-%m-%dT%H:%M:%S%z") # Bookmark is being saved with timezone
+
+                    if stream == state['currently_syncing']:
+
+                        # Check if the interrupted stream has a bookmark written for it
+                        if state["bookmarks"].get(stream):
+                            interrupted_stream_bookmark = self.convert_state_to_utc(state['bookmarks'].get(stream).get('LastUpdatedTime'))
+                            interrupted_bookmark_datetime = dt.strptime(interrupted_stream_bookmark, "%Y-%m-%dT%H:%M:%SZ")
+                        else:
+                            # Assign the start date to the interrupted stream
+                            interrupted_bookmark_datetime = start_date_datetime
```

Comment:
Not sure i understand the purpose of this logic. This block will only only execute against the interrupted `bill_payments` stream, so the reference point should be the state that was set, not the start date.

Reply:
Updated the assertions for the interrupted stream to use the start date as the reference point.